lib/backend/k8s/resources/node_conversion.go

```diff
@@ -49,9 +55,31 @@ func K8sNodeToCalico(node *kapiv1.Node) (*model.KVPair, error) {
 			FelixIPv4:   ip,
 			Labels:      node.Labels,
 			BGPIPv4Addr: ip,
-			BGPIPv6Addr: &net.IP{},
-			BGPASNumber: &asn,
+			BGPASNumber: &bgpAsn,
+			BGPIPv4Net:  bgpV4Cidr,
 		},
 		Revision: node.ObjectMeta.ResourceVersion,
 	}, nil
 }
+
+// Convert a Calico Node to Kubernetes, place BGP configuration info in annotations
+func CalicoToK8sNode(kvp *model.KVPair) (*kapiv1.Node, error) {
+	vals := kvp.Value.(model.Node)
+	annotations := map[string]string {
+		"projectcalico.org/BGPIPv4Addr": vals.BGPIPv4Addr.String(),
+		"projectcalico.org/BGPASNumber": vals.BGPASNumber.String(),
```

Comment:
If there's no `node.BGPASNumber`, then I think we shouldn't be applying the annotation (rather than populating it with a nil value).

Same goes for the other fields.